CHANGELOG.md

```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 - CI now uses flags specified in `Cargo.toml` for `rustdoc` tests.
 - Fixed clippy lints.
+- The `UnionNode<T>` in `pool` now has the correct memory layout for reinterpreting it as a pointer to `T`
```

Comment:
This is not a publicly visible change, readers of the changelog don't need to worry that it may be impacting their code.

```suggestion
- Fixed the memory layout of the internal `UnionNode<T>` type, fixing possible Undefined Behaviour.
```